features/telemetry/adapters/netflow/src/main/java/org/opennms/netmgt/telemetry/adapters/netflow/v5/NetflowHeader.java

```diff
@@ -0,0 +1,96 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017-2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.telemetry.adapters.netflow.v5;
+
+import java.nio.ByteBuffer;
+
+import org.opennms.netmgt.telemetry.adapters.netflow.v5.fields.IntField;
+import org.opennms.netmgt.telemetry.adapters.netflow.v5.fields.LongField;
+import org.opennms.netmgt.telemetry.adapters.netflow.v5.fields.ShortField;
+
+public class NetflowHeader {
+
+    private static final int HEADER_SIZE = 24;
+
+    private static final int BODY_SIZE = 48;
+
+    private final ByteBuffer data;
+
+    public NetflowHeader(ByteBuffer data) {
+        this.data = data;
+    }
+
+    public int getVersion() {
+        return new IntField(0, 1, data, getOffset()).getValue();
+    }
+
+    public int getCount() {
+        return new IntField(2, 3, data, getOffset()).getValue();
```

Comment:
The value may be cached within the `AbstractField` class, but we create a new instance of the field on every get, so we still end up rebuilding the value.

This is fine if we access the value once, but will add unnecessary overhead otherwise.

Reply:
The original prototype worked a bit different, while refactoring it out I missed to not re-instantiate the fields every time. This makes no sense. LoL.